### PR TITLE
Try to solve issue 3417.

### LINKED ISSUE #3417: wrap_comments shouldn't imply format_doc_comments
This is similar to #3209, but a bit worse IMO: currently setting either `normalize_doc_attributes` or `wrap_comments` to `true` implies `format_doc_comments` as well and reformats docs, even if `format_doc_comments = false` is explicitly set in the config.

### PR DESCRIPTION
**Pull request by xiongmao86**
Add two tests first for #3417. Close #3417.

### CODE REVIEW

File: src/comment.rs [hunk 1]
```diff
@@ -656,9 +656,16 @@ impl<'a> CommentRewrite<'a> {
                     _ => {
                         let mut config = self.fmt.config.clone();
                         config.set().wrap_comments(false);
-                        match crate::format_code_block(&self.code_block_buffer, &config) {
-                            Some(ref s) => trim_custom_comment_prefix(&s.snippet),
-                            None => trim_custom_comment_prefix(&self.code_block_buffer),
+                        if config.format_doc_comments() {
```
Comment (xiongmao86) [inline src/comment.rs]: I don't know how to write this if block a better way, please give me suggestion.

Comment (xiongmao86) [reply src/comment.rs]: And I wonder how to select multiply lines in the commit view. please give me suggestion.

**Comment by xiongmao86** [comment]
Setting `normalize_comments` and `wrap_comments` to true no longer imply setting `format_doc_comments`. And I wonder should I change the option `format_doc_comments` to `format_code_in_doc_comments` or just change the description of the option?

**Comment by xiongmao86** [comment]
Travis CI build break because of rand crate has format_doc_comments in rustfmt.toml.

**Comment by scampi** [comment]
The code change looks good to me @xiongmao86 

With the stability guaranty of rustfmt, we can't simply rename the option: rand's build failure is a good example. A possible solution would be to introduce the concept of options "aliases":

- an option, here `format_doc_comments`, can be used with another name, here `format_code_in_doc_comments`
- the first variant of the option's name would be marked as deprecated and removed in a major release.

@topecongiro What do you think of such a mechanism ? Just improving the `format_doc_comments` description could be enough for this issue, but I think a more obvious option's name would be good.

**Comment by RReverser** [comment]
> With the stability guaranty of rustfmt, we can't simply rename the option

All of these are unstable options though, so I'd expect it to be fine to play with naming / values until stabilized?

**Comment by xiongmao86** [comment]
Appveyor build fatal: unable to access 'https://github.com/rust-lang/rustfmt.git/': Could not resolve host: github.com

May be failed because of network accessibility. How can I retry testing?

**Comment by scampi** [comment]
@RReverser ahah indeed ;oD that comment can be discarded then!

**Comment by scampi** [comment]
I believe mentioning the name change in the CHANGELOG should be good enough.

> How to deal with that problem?

Given both options are unstable, nothing needs to be done. The behaviour of `wrap_comments` implying `format_code_in_doc_comment` was wrong to begin with.

**Comment by xiongmao86** [comment]
@RReverser @scampi @topecongiro, Thank you for helping and reviewing my commit. What should I do to get the pull request merge?

**Comment by RReverser** [comment]
Given the discussion above, I'd say rename the option back? (I see you reverted that change)

**Comment by xiongmao86** [comment]
Only rand build in Travis CI failed. I think this is probably good to go. @scampi, @topecongiro, What do you think?

**Comment by topecongiro** [comment]
@xiongmao86 Sorry for the late review. LGTM, thanks!

**Pull request merged**